### PR TITLE
Add interface to get current root and version - Closes #57

### LINKED ISSUE #57: Add interface to get current root and version
### Description
Add interface to return `root` and `version` (rename from `height`) to `StateDB`.
Both values should be save atomically with `commit` and `revert`

### Motivation
- root / version can be used for checking the current state of the stateDB

### Acceptance Criteria
- `root` and `height` should be stored consistently
- Add tests to check `root` and `height` saved with commit

### PR DESCRIPTION
**Pull request by hrmhatef**
### What was the problem?

This PR resolves #57 

### How was it solved?

 - Store `root` and `version` in `rocks-db` also update them in `commit` and  `revert` functions
 - Add a new interface to return `root` and `version`

### How was it tested?
 - Add a new unit test to test new functionality in Rust side
 - Update unit test to check new interface in JS side
 - All tests passed.

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@ use crate::types::{KeyLength, SubtreeHeight, SubtreeHeightKind};
 pub const PREFIX_STATE: &[u8] = &[0];
 pub const PREFIX_SMT: &[u8] = &[1];
 pub const PREFIX_DIFF: &[u8] = &[2];
+pub const PREFIX_CURRENT_STATE: &[u8] = &[3];
```
Comment (matjazv) [inline src/consts.rs:6]: Can all these prefixes be translated into `enum`?

**Review by matjazv — CHANGES_REQUESTED** [review]
Very good implementation!

File: src/state_db.rs [hunk 2]
```diff
@@ -49,6 +57,30 @@ pub struct StateDB {
...
+    fn new(root: &'a [u8], version: BlockHeight) -> Self {
+        Self { root, version }
+    }
+
+    fn to_bytes(&self) -> Vec<u8> {
+        [self.root, &self.version.to_be_bytes()].concat()
+    }
+
+    fn from_bytes(bytes: &'a [u8]) -> Self {
+        if bytes.len() < 4 {
```
Comment (shuse2) [inline src/state_db.rs:70]: why is it 4 here root has 32 bytes?
maybe in this case it could be error?

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: src/state_db.rs (hunk 2 above)
Comment (hrmhatef) [reply src/state_db.rs:70]: Yes, this not necessary, I removed that checking.

File: src/state_db.rs [hunk 3]
```diff
@@ -380,6 +425,28 @@ impl StateDB {
...
+        &self,
+        cb: Root<JsFunction>,
+    ) -> Result<(), mpsc::SendError<options::DbMessage>> {
+        self.common.send(move |conn, channel| {
+            let res: Vec<u8> = conn.get(consts::PREFIX_CURRENT_STATE).unwrap().unwrap();
+            channel.send(move |mut ctx| {
+                let callback = cb.into_inner(&mut ctx);
+                let this = ctx.undefined();
+                let object = ctx.empty_object();
+                let current_state_info = CurrentState::from_bytes(&res);
```
[the remark this answers is not in the document]
Comment (hrmhatef) [reply src/state_db.rs:439]: Done.

File: src/consts.rs (hunk 1 above)
Comment (hrmhatef) [reply src/consts.rs]: I guess there is no a good approach to change them into `enum` but I found a solution for struct implementation, I changed that and push it in separate `commit`, you can check it now.

**Review by matjazv — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**